addon/components/image-cropper.js

```diff
@@ -38,53 +39,29 @@ export default Ember.Component.extend({
   dragEnd: null,
   zoomin: null,
   zoomout: null,
-
   //initialize cropper on did insert element
-  initImageCropper: Ember.on('didRender', function() {
-    var _this = this;
-      _this.$(_this.get('cropperContainer')).cropper({
-        aspectRatio: _this.get('aspectRatio'),
-        crop: _this.get('crop'),
-        preview: _this.get('previewClass'),
-        viewMode: _this.get('viewMode'),
-        dragMode: _this.get('dragMode'),
-        responsive: _this.get('responsive'),
-        center: _this.get('center'),
-        checkCrossOrigin: _this.get('checkCrossOrigin'),
-        toggleDragModeOnDblclick: _this.get('toggleDragModeOnDblclick'),
-        background: _this.get('background'),
-        modal: _this.get('modal'),
-        guides: _this.get('guides'),
-        highlight: _this.get('highlight'),
-        autoCrop: _this.get('autoCrop'),
-        autoCropArea: _this.get('autoCropArea'),
-        dragDrop: _this.get('dragDrop'),
-        movable: _this.get('movable'),
-        resizable: _this.get('resizable'),
-        zoomable: _this.get('zoomable'),
-        zoomOnWheel: _this.get('zoomOnWheel'),
-        zoomOnTouch: _this.get('zoomOnTouch'),
-        cropBoxMovable: _this.get('cropBoxMovable'),
-        cropBoxResizable: _this.get('cropBoxResizable'),
-        rotateable: _this.get('rotateable'),
-        minContainerWidth: _this.get('minContainerWidth'),
-        minContainerHeight: _this.get('minContainerHeight'),
-        minCropBoxWidth: _this.get('minCropBoxWidth'),
-        minCropBoxHeight: _this.get('minCropBoxHeight'),
-        build: _this.get('build'),
-        built: _this.get('built'),
-        dragStart: _this.get('dragStart'),
-        dragMove: _this.get('dragMove'),
-        dragEnd: _this.get('dragEnd'),
-        zoomin: _this.get('zoomin'),
-        zoomout: _this.get('zoomout')
-      });
-  }),
-
-  teardownCropper: Ember.on('willDestroyElement', function() {
-    let cropperElment = this.$(this.get('cropperContainer'));
-    if(cropperElment.data('cropper')) {
-      cropperElment.cropper('destroy');
+  didInsertElement(){
```

Comment:
didRender runs every time something changes in the view, because now we instantiate a new cropperjs object instead of using the jquery version, the code must run only once.

i'm gonna make the changes.

Reply:
👋 @rwwagner90 @0000marcell Hi

What if the image is not rendered at the component initialization (like because we do not have the image to render yet) but only later?

The previous implementation supported this scenario. But now the Cropper object creation fails (no image is found) and this is not retried when the image is later available and rendered.

Reply:
No because the image to crop is sent to CropperJs via the image element. It is part of the template so the template will be re-rendered when the image change.